File: backend/src/main/java/com/mini/credit/dto/credit/CreditContratResponse.java
package com.mini.credit.dto.credit;

import com.mini.credit.enums.StatutCredit;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Getter
@Builder
public class CreditContratResponse {
    private Long creditId;
    private String numeroCredit;
    private Long demandeCreditId;
    private String numeroDemande;
    private StatutCredit statut;

    private String membreNomComplet;
    private String membreAdresse;
    private String membreTelephone;

    private BigDecimal montantAccorde;
    private String devise;
    private BigDecimal tauxInteret;
    private Integer dureeValeur;
    private String dureeUnite;
    private LocalDate datePret;
    private LocalDate dateDecaissement;
    private String objetCredit;
    private String gagePropose;

    private BigDecimal garantieRegleMontant;
    private String garantieRegleLibelle;
    private BigDecimal montantGarantieRequis;
    private BigDecimal montantGarantieBloque;
    private String sourceGarantie;
    private String garantiesMateriellesAcceptees;
    private String penaliteRetardLibelle;

    private BigDecimal totalPrincipal;
    private BigDecimal totalInteret;
    private BigDecimal totalAPayer;

    private String antenneNom;
    private String agentTerrainNom;
    private String gestionnaireNom;
    private String controleurNom;
    private String chefBureauNom;
    private String caissierNom;

    private List<CreditDetailResponse.EcheanceDetail> echeancier;
}

File: frontend/src/app/features/credit/models/credit-contrat-response.ts
import { CreditEcheanceDetail } from './credit-detail-response';
import { StatutCredit } from './enums/statut-credit.enum';

export interface CreditContratResponse {
  creditId: number;
  numeroCredit: string;
  demandeCreditId?: number | null;
  numeroDemande?: string | null;
  statut: StatutCredit;
  membreNomComplet?: string | null;
  membreAdresse?: string | null;
  membreTelephone?: string | null;
  montantAccorde: number;
  devise?: string | null;
  tauxInteret?: number | null;
  dureeValeur?: number | null;
  dureeUnite?: string | null;
  datePret?: string | null;
  dateDecaissement?: string | null;
  objetCredit?: string | null;
  gagePropose?: string | null;
  garantieRegleMontant?: number | null;
  garantieRegleLibelle?: string | null;
  montantGarantieRequis?: number | null;
  montantGarantieBloque?: number | null;
  sourceGarantie?: string | null;
  garantiesMateriellesAcceptees?: string | null;
  penaliteRetardLibelle?: string | null;
  totalPrincipal?: number | null;
  totalInteret?: number | null;
  totalAPayer?: number | null;
  antenneNom?: string | null;
  agentTerrainNom?: string | null;
  gestionnaireNom?: string | null;
  controleurNom?: string | null;
  chefBureauNom?: string | null;
  caissierNom?: string | null;
  echeancier: CreditEcheanceDetail[];
}
